WooCommerce/Classes/ViewRelated/Orders/Order Creation/NotesSection/OrderNotesSection.swift

```diff
@@ -0,0 +1,126 @@
+import SwiftUI
+
+struct OrderNotesSection: View {
+
+    /// Parent view model to access all data
+    @ObservedObject var viewModel: NewOrderViewModel
+
+    /// View model to drive the view content
+    private var notesDataViewModel: NewOrderViewModel.CustomerNoteDataViewModel {
+        viewModel.customerNoteDataViewModel
+    }
+
+    @State private var showEditNotesView: Bool = false
+
+    var body: some View {
+        OrderNotesSectionContent(viewModel: notesDataViewModel, showEditNotesView: $showEditNotesView)
+            .sheet(
+                isPresented: $showEditNotesView,
+                onDismiss: {
+                    viewModel.noteViewModel.userDidCancelFlow()
+                    viewModel.onOrderNoteUpdate()
+                },
+                content: {
+                    EditCustomerNote(
+                        dismiss: {
+                            showEditNotesView.toggle()
+                            viewModel.onOrderNoteUpdate()
+                        },
+                        viewModel: viewModel.noteViewModel
+                    )
+                }
+            )
+    }
+}
+
+private struct OrderNotesSectionContent: View {
+    /// View model to drive the view content
+    var viewModel: NewOrderViewModel.CustomerNoteDataViewModel
+
+    @Binding var showEditNotesView: Bool
+
+    @Environment(\.safeAreaInsets) var safeAreaInsets: EdgeInsets
+
+    var body: some View {
+        VStack(alignment: .leading, spacing: .zero) {
+            HStack(alignment: .top) {
+                Text(Localization.notes)
+                    .headlineStyle()
+                Spacer()
+                if viewModel.customerNotes.isNotEmpty {
+                    createEditNotesButton
+                }
+            }.padding([.leading, .top, .trailing])
+
+            if viewModel.customerNotes.isEmpty {
+                createOrderNotesView
+            } else {
+                createNoteDataView
```

Comment:
Nit: When I see the verb "create" I immediately think that this should be a function because is performing some operations. In contrast, computed variables are just accessing data. 

😅

Reply:
What I mean is that I think this should be `createOrderNotesView()` and `createNoteDataView()` 😅

Reply:
Makes perfect sense! I'm still wrapping my head about how SwiftUI perfectly supports what's expected from a Swift code, it's fixed now! Thanks!